qubesmanager/qube_manager.py

```diff
@@ -410,12 +471,32 @@ def __init__(self, qt_app, qubes_app, dispatcher, parent=None):
         dispatcher.add_handler('property-load',
                                self.on_domain_changed)
 
+        # It needs to store threads until they finish
+        self.threads_list = []
+        self.progress = None
+
         # Check Updates Timer
         timer = QtCore.QTimer(self)
         timer.timeout.connect(self.check_updates)
         timer.start(1000 * 30) # 30s
         self.check_updates()
 
+    def clear_threads(self):
+        for thread in self.threads_list:
+            if thread.isFinished():
+                if self.progress:
+                    self.progress.hide()
+                    self.progress = None
+
+                if thread.msg:
+                    (title, msg) = thread.msg
+                    QtGui.QMessageBox.warning(
+                        None,
+                        self.tr(title),
+                        self.tr(msg))
+
+                self.threads_list.remove(thread)
```

Comment:
Modifying the list while iterating over it is a bad idea. Iterate over its copy.